### ISSUE TITLE
Conan export-pkg fully captures information for package_id in previous install

### ISSUE THREAD

**Issue by memsharded** [conan-io/conan#3769]
https://github.com/conan-io/conan/pull/3768 allows using ``conan export-pkg --profile=xxxx`` even when there is previous ``conan install`` that captured *conaninfo.txt* to account for incomplete information. Issue is: https://github.com/conan-io/conan/issues/3367

It is likely that we want to capture the package information completely in the ``conan install`` so it is easier to obtain later the package_id to do the ``export-pkg``. This is related to other local develoment flow commands.

**Comment by marausch** [comment]
Hello,

I have a case that might relate to this.

By executing the sequence `conan install/build/export-pkg` (instead of `conan create`) two _conaninfo.txt_ files are created.
The first one in the build directory has following content (empty lines removed):
```
    [full_settings]
        arch=x86_64
        build_type=Debug
        compiler=gcc
        compiler.libcxx=libstdc++11
        compiler.version=4.9
        os=Windows
    [full_requires]
    [full_options]
    [recipe_hash]
    [env]
        CONAN_CMAKE_GENERATOR=Ninja
```
while the second one in the _package_ folder in conan cache contains the following:
```
    full_settings]
        arch=x86_64
        build_type=Debug
        compiler=gcc
        compiler.libcxx=libstdc++11
        compiler.version=4.9
        os=Windows
    [full_requires]
    [full_options]
    [recipe_hash]
        efd4803aea8a060cfb38c1cdd130bd1d
    [env]
```
Apart form the `recipe_hash` expectedly missing in the local file, we can see that the last line (`CONAN_CMAKE_GENERATOR=Ninja`) has diasppeared from the second _conaninfo.txt_ file
(the same file created in _package_ folder by `conan create` obviously contains both lines).

In fact, this doesn't really look like a case of "incomplete information" captured at `conan install`...
Thanks for any enlightenment.

**Comment by memsharded** [comment]
Hi @marausch 

Yes, it is not incomplete information from ``conan install``, but a loss of information when it is exported. I think the new flow that will be released in 1.9, where the default in ``conan export-pkg`` is to specify the profile again, will solve this discrepancy. Then, yes, we will try to automate this, so the full profile is correctly stored and later used for ``conan export-pkg``. Please try the 1.9 flow when it is released and report. Thanks for the feedback!

**Comment by memsharded** [comment]
This was changed in another direction:
- ``conan install`` no longer saves the state, so ``conan build`` also completely builds the graph
- ``conan export-pkg`` will keep building the graph from inputs, which will be necessary and they should match the inputs given at ``conan install``
